spec/affinity_awareness/AffinityAwarenessMultipleConnections.spec.js

```diff
@@ -0,0 +1,202 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+require('jasmine-expect');
+
+const TestingHelper = require('../TestingHelper');
+const AffinityAwarenessTestUtils = require('./AffinityAwarenessTestUtils');
+const IgniteClient = require('@gridgain/thin-client');
+const ObjectType = IgniteClient.ObjectType;
+
+const CACHE_NAME = '__test_cache';
+const CUSTOM_AFFINITY_CACHE = 'custom-affinity';
+const PARTITIONED0_CACHE = 'partitioned0';
+const PARTITIONED1_CACHE = 'partitioned1';
+const PARTITIONED3_CACHE = 'partitioned3';
+const REPLICATED_CACHE = 'replicated';
+const SERVER_NUM = 3;
+
+describe('affinity awareness multiple connections test suite >', () => {
+    let igniteClient = null;
+
+    beforeAll((done) => {
+        Promise.resolve().
+            then(async () => {
+                await TestingHelper.init(true, SERVER_NUM, true);
+                igniteClient = TestingHelper.igniteClient;
+                await testSuiteCleanup(done);
+            }).
+            then(done).
+            catch(error => done.fail(error));
+    }, TestingHelper.TIMEOUT);
+
+    afterAll((done) => {
+        Promise.resolve().
+            then(async () => {
+                await testSuiteCleanup(done);
+                await TestingHelper.cleanUp();
+            }).
+            then(done).
+            catch(_error => done());
+    }, TestingHelper.TIMEOUT);
+
+    it('all cache operations with affinity awareness and multiple connections', (done) => {
+        Promise.resolve().
+            then(async () => {
+                const cache = await getOrCreateCache(ObjectType.PRIMITIVE_TYPE.INTEGER, ObjectType.PRIMITIVE_TYPE.INTEGER);
+                await AffinityAwarenessTestUtils.testAllCacheOperations(cache);
+            }).
+            then(done).
+            catch(error => done.fail(error));
+    });
+
+    it('all cache operations with affinity awareness and bad affinity', (done) => {
+        Promise.resolve().
+            then(async () => {
+                const cache = await getOrCreateCache(ObjectType.PRIMITIVE_TYPE.INTEGER, ObjectType.PRIMITIVE_TYPE.INTEGER, CUSTOM_AFFINITY_CACHE);
+                await AffinityAwarenessTestUtils.testRandomNode(igniteClient, cache);
+            }).
+            then(done).
+            catch(error => done.fail(error));
+    });
+    
+    it('put with affinity awareness and unknown cache', (done) => {
+        Promise.resolve().
+            then(async () => {
+                const cache = await getCache(ObjectType.PRIMITIVE_TYPE.INTEGER, ObjectType.PRIMITIVE_TYPE.INTEGER, '__unknown_cache_359f72tg');
+                let key = 42;
+                try {
+                    await cache.put(key, key);
+                }
+                catch (error) {
+                    expect(error.toString()).toContain('Cache does not exist');
```

Comment:
Same as above - let's make sure `error.message` is good

Reply:
Fixed